File: app/src/main/java/com/example/myapplication/ServerInterface.java
package com.example.myapplication;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.*;

public interface ServerInterface {

    @GET("/users/0")
    Call<TokenResponse> connectivityCheck();



    @GET("/users/{user_id}")
    Call<TokenResponse> getUser(@Path("user_id") String userId);


    @GET("/todos")
    Call<List<TokenResponse>> getAllTicketsForUser(@Query("user_id") String userId);


    // the result will be the ticket as was created in the server (e.g. change in the "id" field, etc)
    @POST("/todos")
    Call<TokenResponse> insertNewTicket(@Body TokenResponse token);

}

File: app/src/main/java/com/example/myapplication/ServerInterface.kt
package com.example.myapplication

import retrofit2.Call
import retrofit2.http.*

data class SetUserPrettyNameRequest(
    var prettyName: String?
)

interface ServerInterface {
    @GET("/users/{user_name}/token/")
    fun getToken(@Path("user_name") userName: String): Call<TokenResponse>

    @GET("/user/")
    fun getUser(@Header("Authorization") auth: String): Call<UserResponse>

    @Headers("Content-Type: application/json")
    @POST("/user/edit/")
    fun setUserPrettyName(@Header("Authorization") auth: String,
                          @Body request: SetUserPrettyNameRequest) : Call<UserResponse>
}